### DISCUSSION TITLE
PrimeVue DataTable TS2322: Type 'string' is not assignable to type 'object'.

### DISCUSSION THREAD

**Discussion by crosswins** [primefaces/community#1357] [PrimeVue] [answered]
Just today I started getting the above error which is caused by tableStyle on a DataTable. The project is Vue Composition app with TypeScript. The WebStorm TS compiler does not complain about an error only the npm production build

The build error is as follows:

✓ built in 9.59s
src/components/complaint/ComplaintAllocations.vue:9:14 - error TS2322: Type 'string' is not assignable to type 'object'.
9              tableStyle="min-width: 50rem"
               ~~~~~~~~~~
  node_modules/primevue/datatable/DataTable.d.ts:1120:5
    1120     tableStyle?: object | undefined;
             ~~~~~~~~~~
    The expected type comes from property 'tableStyle' which is declared here on type 'DataTableProps & VNodeProps & AllowedComponentProps & ComponentCustomProps & Record<string, unknown>'

The markup that is causing this is as follows:
`<DataTable :value="complaintAllocations"
             sort-field="commentDate"
             :sort-order="1"
             dataKey="commentDate"
             :metaKeySelection="false"
             contextMenu
             striped-rows
             tableStyle="min-width: 50rem"
             paginator :rows="5"
             :rowsPerPageOptions="[5, 10, 20, 50]"
             :alwaysShowPaginator="complaintAllocations?.value?.length > 0" >
    <Column field="costCentre" sortable header="Cost Centre" />
    <Column field="amount" sortable header="Amount" />
    <Column field="costCentreDescription" sortable header="Description" />
  </DataTable >`

My package.json is as follows:
`{
  "name": "vue-app-vanilla-comp",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "run-p type-check build-only",
    "preview": "vite preview",
    "build-only": "vite build",
    "type-check": "vue-tsc --noEmit -p tsconfig.app.json --composite false",
    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore",
    "format": "prettier --write src/"
  },
  "dependencies": {
    "@fortawesome/fontawesome-free": "^6.4.2",
    "@fortawesome/fontawesome-free-solid": "^5.0.13",
    "@fortawesome/vue-fontawesome": "^3.0.3",
    "@vee-validate/yup": "^4.12.3",
    "axios": "^1.5.0",
    "chart.js": "3.3.2",
    "moment": "^2.29.4",
    "pinia": "^2.1.6",
    "primeflex": "^3.3.1",
    "primeicons": "^6.0.1",
    "primevue": "^3.38.1",
    "prismjs": "^1.29.0",
    "sass": "^1.69.5",
    "vee-validate": "^4.12.3",
    "vue": "^3.3.4",
    "vue-router": "^4.2.4",
    "vue2-dropzone": "^3.6.0",
    "vuex": "^4.1.0",
    "yup": "^1.3.3"
  },
  "devDependencies": {
    "@rushstack/eslint-patch": "^1.3.2",
    "@tsconfig/node18": "^18.2.0",
    "@types/node": "^18.17.5",
    "@vitejs/plugin-vue": "^4.3.1",
    "@vue/eslint-config-prettier": "^8.0.0",
    "@vue/eslint-config-typescript": "^11.0.3",
    "@vue/tsconfig": "^0.4.0",
    "eslint": "^8.46.0",
    "eslint-plugin-vue": "^9.16.1",
    "npm-run-all": "^4.1.5",
    "prettier": "^3.0.0",
    "typescript": "~5.1.6",
    "vite": "^4.5.2",
    "vue-tsc": "^1.8.8"
  }
}
`

Can anyone see what may be wrong?

**Comment by tugcekucukoglu** [comment, marked as answer]
Thanks for the report. It will be fixed for the next version.
